File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n = prices.size();
        vector<int> accumulate(n);
        accumulate[0] = 0;
        int a = 0;
        for(int i=1; i< n; i++ ){
            a = accumulate[i-1] + prices[i] - prices[i-1];
            accumulate[i] = max(0, a);
        }
        int max = 0;
        for(int i = 0; i<n ; i++){
            if(accumulate[i] > max){max = accumulate[i];}
        }
        return max;
    }
};

File: Data Structures & Algorithms/balanced-binary-tree/submission-0.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution {
    public bool IsBalanced(TreeNode root) {
        if(root is null) return true;
        bool isItBalenced = true;
        int _hight = hight(root, ref isItBalenced);
        return isItBalenced;
    }

    public int hight(TreeNode root, ref bool isItBalenced){
        if(root is null) return 0;
        int rightHight = hight(root.right, ref isItBalenced);
        int leftHight = hight(root.left, ref isItBalenced);
        bool isItBalencedLocal = Math.Abs(rightHight - leftHight) > 1 ? false : true;
        if (isItBalencedLocal == false) isItBalenced = false;
        return 1 + Math.Max(rightHight , leftHight);
    }
}
